spacy/matcher/dependencymatcher.pyx

```diff
@@ -424,18 +414,10 @@ cdef class DependencyMatcher:
         return []
 
     def _right_sib(self, doc, node):
-        candidate_children = []
-        for child in list(doc[node].head.children):
-            if child.i > node:
-                candidate_children.append(doc[child.i])
-        return candidate_children
+        return [doc[child.i] for child in doc[node].head.children if child.i > node]
 
     def _left_sib(self, doc, node):
-        candidate_children = []
-        for child in list(doc[node].head.children):
-            if child.i < node:
-                candidate_children.append(doc[child.i])
-        return candidate_children
+        return [doc[child.i] for child in doc[node].head.children if child.i < node]
```

Comment:
I'm trying to think why `doc[node].head.children` (which is a generator) was converted to a `list` first in the original code, but I think this should be fine as well?

Reply:
`list(...)` simply exhausts whatever iterable, generator or otherwise, one passes to it, saving the values to a list. In this circumstance, we aren't depending on any property of `doc[node].head.children` except that it is iterable, so converting it to a list ahead of the list comp has no effect on the result.